src/GraphQL.Conventions/Web/Response.cs

```diff
@@ -26,33 +28,26 @@ public class Response
             ValidationResult = result;
         }
 
-        public Request Request { get; private set; }
+        public Request Request { get; }
 
         public string QueryId => Request?.QueryId;
 
-        public ExecutionResult ExecutionResult { get; private set; }
+        public ExecutionResult ExecutionResult { get; }
 
-        public Validation.IValidationResult ValidationResult { get; private set; }
+        public Validation.IValidationResult ValidationResult { get; }
 
-        public string Body
+        public string Body => GetBodyAsync().GetAwaiter().GetResult();
```

Comment:
Consder to remove sync API or mark it obsolete

Reply:
I wouldn't mind removing the sync ones. It would be a breaking change but as GraphQL.Net 3.0 already brings his lot of breaking changes it's not such a big deal and upgrading the code is quite straight forward. Do you have any version compatibility strategy ?

But if we do so it would be interesting to convert other method to async for consistency:
   - GraphQLEngine.Execute: rename to ExecuteAsync
   - IGraphQLExecutor.Execute: rename to ExecuteAsync
   - IGraphQLExecutor.Validate: convert to Async
   - IRequestHandler.ProcessRequest: rename to ProcessRequestAsync
   - IRequestHandler.Validate: convert to Async
   - IRequestHandler.DescribeSchema: convert to Async
   - Response.Body: remove property (as per you other comment).

What do you think?

Reply:
> Do you have any version compatibility strategy ?

Well, graphql-dotnet version 3 (next major) is already backward incompatible in many places, but I don’t see a problem in that. After the release, all dependent packages will need to release their major versions as well, that's all.

> What do you think?

I think that changing the signature of some methods to the asynchronous version is perfectly normal. See graphql-dotnet/graphql-dotnet#1371

Reply:
I note that the server project is also in the preview version using the preview of the base library. I can’t say that the preview version of graphql-dotnet is less stable than 2.4, rather the opposite.

Reply:
I created a separate PR #195